src/log.rs

```diff
@@ -0,0 +1,106 @@
+//! Logging adapter for Laminar
+//!
+//! This module implements a simple, threaded-logger-friendly logging adapter. Logging adapters are
+//! used to attach an arbitrary logger into Laminar.
+use std::fmt;
+use std::sync::Arc;
+
+/// Logger trait for laminar
+///
+/// Any user of Laminar can implement this trait to attach their favorite logger to an instance of
+/// laminar. The log levels correspond to the same log levels as in the `log` crate.
+pub trait LaminarLogger {
+    /// Log a trace message
+    fn trace(&self, disp: Displayer);
+    /// Log a debug message
+    fn debug(&self, disp: Displayer);
+    /// Log an info message
+    fn info(&self, disp: Displayer);
+    /// Log a warning message
+    fn warn(&self, disp: Displayer);
+    /// Log an error message
+    fn error(&self, disp: Displayer);
+}
+
+// ---
+
+/// Holds a handle to a formatter function while implementing the [fmt::Display] trait.
+pub struct Displayer {
+    data: Arc<dyn Fn(&mut ::std::fmt::Formatter) -> ::std::fmt::Result + Send + Sync>,
```

Comment:
`Arc` is most general, for instance if you want to send your log messages to another thread.
To be honest I'd prefer if we can do it completely without dynamic allocation but that doesn't allow for arbitrary data to be sent (because sized information is required for that) - data sent through a `move |f| format!(f, "{}", x)` requires it because of the arbitrary size of `x`.

Reply:
Is there an actual need/use-case of wanting to do that today? I prefer not using the most general solution unless we have a hard requirement that it is needed.